### PR TITLE
[SR-7629] Fix wrong fixit when the type doesn't conform to a public protocol

### PR DESCRIPTION
**Pull request by naru-jpn**
Wrong fixit occurred when initializer is memberwiseInitializer. So I disabled fixit only in that case.

Resolves [SR-7629](https://bugs.swift.org/browse/SR-7629).

### CODE REVIEW

**Comment by naru-jpn** [comment]
I wrote tests and checked to pass all tests on my PC.

**Comment by jrose-apple** [comment]
Brent, if you're good with this version so am I; feel free to run tests and merge when you're satisfied with it. (Though I'm sad `isImplicit` isn't good enough.)

File: lib/Sema/TypeCheckProtocol.cpp [hunk 1]
```diff
@@ -2773,6 +2773,12 @@ ConformanceChecker::resolveWitnessViaLookup(ValueDecl *requirement) {
                        requiredAccess,
                        protoAccessScope.accessLevelForDiagnostics(),
                        proto->getName());
+        if (auto *decl = dyn_cast<AbstractFunctionDecl>(witness)) {
+          auto isMemberwiseInitializer = decl->getBodyKind() == AbstractFunctionDecl::BodyKind::MemberwiseInitializer;
```
Comment (xwu) [inline lib/Sema/TypeCheckProtocol.cpp]: Minor nit: can you clang-format this change?

**Comment by jrose-apple** [comment]
@brentdax, any further comments?

@swift-ci Please test

**Comment by naru-jpn** [comment]
@jrose-apple @brentdax Are there other things that you think I should do?

**Comment by jrose-apple** [comment]
I guess we can take this, and if Brent has further follow-up comments he can still provide them after the merge.

**Pull request merged**